src/main/java/randoop/condition/SpecificationTranslator.java

```diff
@@ -0,0 +1,398 @@
+    if (accessibleObject instanceof Method) {
+      Method method = (Method) accessibleObject;
+      // get condition method signatures
+      preconditionSignature = getPreConditionSignature(method);
+      postconditionSignature = getPostConditionSignature(method);
+
+      // get condition method parameter declaration strings
+      parameterNames.add(identifiers.getReceiverName());
+      parameterNames.addAll(identifiers.getParameterNames());
+      preConditionDeclarations = preconditionSignature.getDeclarationArguments(parameterNames);
+      parameterNames.add(identifiers.getReturnName());
+      postConditionDeclarations = postconditionSignature.getDeclarationArguments(parameterNames);
+    } else if (accessibleObject instanceof Constructor) {
+      Constructor<?> constructor = (Constructor) accessibleObject;
+      // get condition method signatures
+      preconditionSignature = getPreConditionSignature(constructor);
+      postconditionSignature = getPostConditionSignature(constructor);
+
+      // get condition method parameter declaration strings
+      parameterNames.addAll(identifiers.getParameterNames());
+      preConditionDeclarations = preconditionSignature.getDeclarationArguments(parameterNames);
+      parameterNames.add(identifiers.getReturnName());
+      postConditionDeclarations = postconditionSignature.getDeclarationArguments(parameterNames);
+    } else {
+      throw new RandoopConditionError("Specification operation is neither a method or constructor");
+    }
+    NameReplacementMap replacementMap = createReplacementMap(parameterNames);
+    return new SpecificationTranslator(
+        preconditionSignature,
+        preConditionDeclarations,
+        postconditionSignature,
+        postConditionDeclarations,
+        replacementMap,
+        compiler);
+  }
+
+  /**
+   * Create the {@link RawSignature} for the condition method for evaluating a precondition for the
+   * given method or constructor.
+   *
+   * <p>The parameter types of the RawSignature are the declaring class as the receiver type
+   * followed by the parameter types of {@code executable}.
+   *
+   * <p>Note: The declaring class of the condition method is actually determined by {@link
+   * ConditionMethodCreator#createMethod(RawSignature, String, String, SequenceCompiler)}
+   *
+   * @param executable the method or constructor to which the precondition belongs
+   * @return the {@link RawSignature} for a pre-condition method of {@code executable}
+   */
+  private static RawSignature getPreConditionSignature(Executable executable) {
+    boolean isMethod = executable instanceof Method;
+    Class<?> declaringClass = executable.getDeclaringClass();
+    Class<?>[] parameterTypes = executable.getParameterTypes();
+    String packageName = getPackageName(declaringClass.getPackage());
+    return ConditionMethodCreator.getRawSignature(
+        packageName, declaringClass, parameterTypes, null, isMethod);
```

Comment:
The `isMethod` argument passed to the `firstArgumentIsReceiver` formal parameter looks wrong here and for `getPostConditionSignature`.  (The two methods differ by just one line; I would unify them to avoid code duplication.)
An inner class constructor has a receiver.
If the current code is correct, then document why my reaction was incorrect.